src/java.xml/share/classes/com/sun/java_cup/internal/runtime/lr_parser.java

```diff
@@ -149,6 +149,7 @@ public abstract class lr_parser {
     private int grpCount = 0;
     private int opCount = 0;
     private int totalOpCount = 0;
+    private int lastSym;
```

Comment:
`lastSym` is never initialized. It's OK for the first time, but should this be reset for the next use (if any), e.g. somewhere around line 595?

Reply:
Right, -1 would be appropriate as 0 indicates EOF.